web/frontend/project-checklist.md

```diff
@@ -0,0 +1,155 @@
+  - What will you use for running scripts and managing dependencies (npm or yarn)?
+  - Will you use a JS framework (React, vue, etc.)?
+- For CSS:
+  - How do you want to write CSS?
+  - What classname conventions do you want to use? (BEM, OOCSS)
+  - Normalize (resets native browser CSS to one standard)
+  - Autoprefixer (automatically adds browser prefixes to newer CSS features that may not have consistent support or syntax)
+  - Pre-/post-compilers (SASS/SCSS, LESS, PostCSS)
+  - CSS modules
+  - CSS-in-JS (styled components, etc.)
+
+### Additional considerations if building a SPA
+
+- [ ] Global state management (Redux, context, etc.)
+  - [ ] side effects (sagas, thunks, hooks?)
+- [ ] Client-side routing
+- [ ] Managing head/meta/title tags
+- [ ] XHR/fetch service
+  - What kind of API(s) do you need to interact with? (REST, GraphQL)
+- [ ] Static asset loading (HTML, CSS, images, fonts, SVGs, etc.)
+
+### User Interaction
+
+- Forms
+  - Validating
+  - Submitting
+  - Showing output
+- File uploads
+
+### Managing & Displaying Content
+
+- Static content from the code
+  - i18n
+- Dynamic content from a CMS, API, etc.
+- User-generated content (UGC)
+  - Escaping entities
+  - Formatting
+
+### Maintaining code standards & developer tools
+
+- [ ] Linting/formatting (eslint, prettier, stylelint)
+- [ ] Commit hooks (husky)
+- Browser dev tools
+- React developer tools
+- Redux debugger
+
+### Working with Design
+
+- Pre-fab frameworks (i.e., Bootstrap, MaterialUI, etc.)
+- Component-driven development
+  - Maintain component libraries for documentation and testing (i.e., Storybook)
+  - Build and view UI components outside the context of the application (don't need to connect to API, database, etc.)
+  - Iterate on UI elements over time
+  - Automate cross-browser/device testing by using this with visual testing tools (see below)
+- How do you want to render icons? SVGs? FontAwesome?
+
+## Testing
+
+- Unit tests
+  - Test small pieces or components in isolated environments
+  - Good to run continuously during development, as well as part of CI checks
+  - Jest
+- Integration tests
+  - Test larger chunks of the app and how they work together with partial or complete context
+  - Good to run continuously during development, as well as part of CI checks
+  - Jest
+- End-to-end (E2E) tests
+  - Test complete user flows in the browser, as if you were a user
+  - Most often useful for high-level regression tests and critical paths
+  - Testing in different environments (browsers, devices) can be useful at this point
+  - Heavy load, often best done in CI
+  - Good to run during development for high-level changes such as refactors, changing URLs or entrypoints
+  - Cypress, Saucelabs, Browserstack
+- Visual testing
+  - Scripts that crawl through URLs or component libraries and take screenshots
+  - Tests pass or fail when the images are compared to references
+  - Heavy load, often best done in CI
+  - Happo, Loki, Percy
+
```

Comment:
Accessibility Testing
- Can we done through unit/integration testing or end to end.  
- pa11y, axe-core